__tests__/utils/bigint.test.js.ts

```diff
@@ -54,18 +54,41 @@ const bigIntObjSchema = z.object({
 
 describe('test JSONBigInt', () => {
   test('should parse numbers', () => {
+    // Doubles should be parsed normally as JS Numbers.
     expect(JSONBigInt.parse('123')).toStrictEqual(123);
     expect(JSONBigInt.parse('123.456')).toStrictEqual(123.456);
     expect(JSONBigInt.parse('1.0')).toStrictEqual(1);
     expect(JSONBigInt.parse('1.000000000000')).toStrictEqual(1);
-    expect(JSONBigInt.parse('12345678901234567890')).toStrictEqual(12345678901234567890n);
-    expect(JSONBigInt.parse('12345678901234567890.000')).toStrictEqual(12345678901234567890n);
     expect(JSONBigInt.parse('1e2')).toStrictEqual(100);
     expect(JSONBigInt.parse('1E2')).toStrictEqual(100);
 
-    expect(() => JSONBigInt.parse('12345678901234567890.1')).toThrow(
-      Error('large float will lose precision! in "12345678901234567890.1"')
-    );
```

Comment:
Why did you remove this?

Reply:
This was actually kind of incorrect. The JSON specification imposes no limitation on numbers in JSON files, that is, both very large integers and very large floats are allowed. So in theory we could have `"some_num": 12345678901234567890.1` in a JSON.

However, in practice, JS would try to parse this as a `Number` and Python would try to parse this as a `float`, and both are represented using the IEEE 754 double specification. This means this number is actually useless in all cases, as it's impossible to represent it precisely as a JS `Number` or Python `float`. Neither language would ever generate a JSON with it.

Check this out, in JS and Python:

```js
> JSON.parse('12345678901234567890.1') === 12345678901234567000.0
true
```

```python
>>> json.loads('12345678901234567890.1') == 12345678901234567000.0
True
```

Summing up, I was unnecessarily prohibiting this parse, while we should actually just let it be parsed normally as a double by JS and Python. In practice, this number will never even appear in a JSON generated by these languages.